### DISCUSSION TITLE
Import and export CSV [Feature request]

### DISCUSSION THREAD

**Discussion by csfalcao** [thanhlmm/refi-app#42] [Ideas]
Import and export CSV

**Comment by thanhlmm** [comment]
Hi @csfalcao, thanks for your feedback and interest in Refi App.
You can import CSV file by Press `Cmd + I` or Search for Import feature with `Cmd + Shift P`

![image](https://user-images.githubusercontent.com/9281080/151733368-09d16ad8-50ff-4f3b-bbac-b5b8dc6ad7cd.png)

**Comment by csfalcao** [comment]
That's amazing! Well, I didn't find that info in the documentation or in the features section on the site. I think it's important to make those features on sight, maybe a "?" icon can help.

**Comment by csfalcao** [comment]
Hi. The import feature didn't work for me. I tried to upload a simple 2 columns Excel base CSV and RefiApp won't recognize it. I convert it to a JSON file in the [CSV2JSON](https://csvjson.com/csv2json) website tool and the JSON file is recognized (the button in RefiApp displays "import 204 doc(s)") but when I click it won't do anything. Windows version here. Please, could you check it out?

**Reply by csfalcao** [reply]
[hymns_list_comma.CSV](https://github.com/thanhlmm/refi-app/files/7982411/hymns_list_comma.CSV)

**Reply by csfalcao** [reply]
Github won't allow json files, so I zipped
[hymns_list_comma.zip](https://github.com/thanhlmm/refi-app/files/7982413/hymns_list_comma.zip)

**Comment by thanhlmm** [comment]
Hey @csfalcao the new version is released which fixes the above issue.
Check it out https://github.com/thanhlmm/refi-app/releases/tag/v0.0.13

**Comment by csfalcao** [comment]
The CSV file still is not recognizable. The JSON import now works. The file is recognized (the button changes to "import 204 docs") and the import was instant, great!

Some notes: 
1 - unlikely when we add/update/delete in RefiApp, it asks for us to commit before writing to FB. The import was committed in the import.
2 - I had to add the columns by clicking the + button and typing the matching field, as the main view won't update and will display only the documents ID (the big panel on the right displays all info from the documents, see image below) 
3 - the number field were imported as String, is there a way to convert all of them to number/integer type at once?

Thanks

![image](https://user-images.githubusercontent.com/2768615/152697862-10937214-9e7b-4472-af89-83c985078fc1.png)

**Reply by thanhlmm** [reply]
Hi
1. For the CSV issue, I have checked and it is because your CSV have something wrong with encoding the header, so the import fails. I think we can improve by showing some warning before import
2. 
![image](https://user-images.githubusercontent.com/9281080/152709398-2e24dd0a-aa11-485e-a1ea-96498ee7e64c.png)
You can always hit `Reset` button and the app will show the column automatically
3.
I believe that you can always do that with JSON import. If the type is number, the imported record should be number. Also I plan to have a terminal/code execute area so we can run an update comment right in the app (Like FireFoo) but it gonna take lots of effort to do

**Reply by csfalcao** [reply]
Zapier has a preview window, which shows the first document headers and values. There you can change the field type and hit import. FireFoo can make multi updates using scripting, but no coders like me it can't use it.
